### PR TITLE
Refactor config paths, fix CLI options

### LINKED ISSUE #859: Project arguments are ignored in command line
## Expected Behavior

Using the command line arguments `-P` or `--project` should allow you to specify a clasp project file according to the help text: `path to a project file or to a folder with a '.clasp.json' file`.

## Actual Behavior

These arguments are ignored.

## Steps to Reproduce the Problem

1. Move your `.clasp.json` file to a new name like `myClasp.json`
2. Run `clasp -P myClasp.json versions`
3. You will see an error that `.clasp.json` was not found because the `-P` arg did not properly use the provided file

## Specifications

- Version (`clasp -v`): 2.4.0

### PR DESCRIPTION
**Pull request by sqrrrl**
fix: Honor --project and other CLI options for overriding config file paths

Fixes #859 & #832.  Removes PathProxy as no longer needed -- simpler to inline the path resolution in Config and not have the extra layer of indirection.

### CODE REVIEW

File: src/commands/logout.ts [hunk 1]
```diff
@@ -1,40 +1,16 @@
 import {Conf} from '../conf.js';
-import {DOTFILE} from '../dotfile.js';
-import {hasOauthClientSettings} from '../utils.js';
+import fs from 'fs';
```
Comment (PopGoesTheWza) [inline src/commands/logout.ts:2]: shouldn't it be fs-extra?

File: src/index.ts [hunk 2]
```diff
@@ -52,13 +52,14 @@ import versions from './commands/versions.js';
 import {Conf} from './conf.js';
 import {PROJECT_NAME} from './constants.js';
 import {spinner, stopSpinner} from './utils.js';
+import fs from 'fs';
```
Comment (PopGoesTheWza) [inline src/index.ts:55]: fs-extra?

**Review by PopGoesTheWza — CHANGES_REQUESTED** [review]
@sqrrrl LGTM

I suggest the use of 'fs-extra' over 'fs'.

File: src/commands/logout.ts (hunk 1 above)
Comment (sqrrrl) [reply src/commands/logout.ts]: done

File: src/index.ts (hunk 2 above)
Comment (sqrrrl) [reply src/index.ts]: done

**Review by PopGoesTheWza — APPROVED** [review]
LGTM

FYI, I just published updated `dotf` and `ts2gas` packages.

**Review by PopGoesTheWza — APPROVED** [review]

**Pull request merged**